third_party/cares/cares.BUILD

```diff
@@ -138,10 +144,23 @@ cc_library(
     copts = [
         "-D_GNU_SOURCE",
         "-D_HAS_EXCEPTIONS=0",
-        "-DNOMINMAX",
         "-DHAVE_CONFIG_H",
-    ],
+    ] + select({
+        ":windows": [
+            "-DNOMINMAX",
+            "-DWIN32_LEAN_AND_MEAN",
```

Comment:
I might be missing something here, but since the cmake-generated `ares_config.h` has `#undef DWIN32_LEAN_AND_MEAN`, what's the effect of this?

Reply:
> what's the effect of this?

I think they will cancel out each other.

C-Ares add `-DWIN32_LEAN_AND_MEAN` in [`CMakeLists.txt#L169`](https://github.com/c-ares/c-ares/blob/3be1924221e1326df520f8498d704a5c4c8d0cce/CMakeLists.txt#L169), but then undef it unconditionally in [`ares_config.h.cmake#L419`](https://github.com/c-ares/c-ares/blob/3be1924221e1326df520f8498d704a5c4c8d0cce/ares_config.h.cmake#L419).

`ares_config.h` is only included in `ares_setup.h`. After including `ares_config.h`, `ares_setup.h` defines `WIN32_LEAN_AND_MEAN` once again at [`ares_setup.h#L109`](https://github.com/c-ares/c-ares/blob/a5b2e99207fd12d1e9d0413f4a6cdc8d0d9bdda3/ares_setup.h#L109). (No, I have no idea why C-Ares is doing this macro dance).

I think it is best we just follow what `CMakeLists.txt` does and don't think too much.

Reply:
having trouble seeing how this cancels, can this `-DWIN32_LEAN_AND_MEAN` just be removed?

Reply:
I just tested removing this flag and build a project that depends on gRPC. I see no difference. No warnings at all. Should I just remove this flag?

Reply:
Yeah I'd prefer remove it. I believe it has no effect, and it's one less thing to have

Reply:
Done.